ocs_ci/helpers/helpers.py

```diff
@@ -1729,6 +1763,93 @@ def verify_volume_deleted_in_backend(
+    """
+    Delete an Image/Subvolume in the backend
+
+    Args:
+         img_uuid (str): Part of VolID which represents corresponding
+            image/subvolume in backend, eg:
+            ``oc get pv/<volumeName> -o jsonpath='{.spec.csi.volumeHandle}'``
+            Output is the CSI generated VolID and looks like:
+            ``0001-000c-rook-cluster-0000000000000001-f301898c-a192-11e9-852a-1eeeb6975c91``
+            where image_uuid is ``f301898c-a192-11e9-852a-1eeeb6975c91``
+         pool_name (str): The of the pool
+
+    Returns:
+         bool: True if image deleted successfully
+            False if:
+                Pool not found
+                image not found
+                image not deleted
+
+    """
+    cmd = ""
+    valid_error = []
+    pool_cr = get_pool_cr(pool_name)
+    if pool_cr is not None:
+        if pool_cr["kind"] == "CephFilesystem":
+            interface = "CephFileSystem"
+        else:
+            interface = pool_cr["kind"]
+        logger.info(f"pool {pool_cr} kind is {interface}")
+    else:
+        logger.info(
+            f"Pool {pool_name} has no kind of "
+            f"{constants.CEPHBLOCKPOOL} "
+            f"or {constants.CEPHFILESYSTEM}"
+        )
+        return False
+
+    # Checking if image is present before trying to delete
+    image_present_results = is_volume_present_in_backend(
+        interface=interface, image_uuid=img_uuid, pool_name=pool_name
+    )
+
+    # Incase image is present delete
+    if image_present_results:
+        if interface == constants.CEPHBLOCKPOOL:
+            logger.info(
+                f"Trying to delete image csi-vol-{img_uuid} from pool {pool_name}"
+            )
+            valid_error = ["No such file or directory"]
+            cmd = f"rbd rm -p {pool_name} csi-vol-{img_uuid}"
+
+        if interface == constants.CEPHFILESYSTEM:
+            logger.info(
+                f"Trying to delete image csi-vol-{img_uuid} from pool {pool_name}"
+            )
+            valid_error = [
+                f"Subvolume 'csi-vol-{img_uuid}' not found",
+                f"subvolume 'csi-vol-{img_uuid}' does not exist",
+            ]
+            cmd = f"ceph fs subvolume rm {get_cephfs_name()}" f" csi-vol-{img_uuid} csi"
+
+        ct_pod = pod.get_ceph_tools_pod()
+        try:
+            ct_pod.exec_ceph_cmd(ceph_cmd=cmd, format=None)
+        except CommandFailed as ecf:
+            if any([error in str(ecf) for error in valid_error]):
+                logger.info(
+                    f"Error occurred while verifying volume is present in backend: "
+                    f"{str(ecf)} ImageUUID: {img_uuid}. Interface type: {interface}"
+                )
+                return False
+
+        verify_img_delete_result = is_volume_present_in_backend(
+            interface=interface,
+            image_uuid=img_uuid,
+            pool_name=pool_name
+        )
+        if verify_img_delete_result is False:
```

Comment:
can be : `if not verify_img_delete_result:`

Reply:
Done